lib/swagcov/dotfile.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Swagcov
+  class BadConfigurationError < RuntimeError
+  end
+
+  class Dotfile
+    def initialize
+      @dotfile = YAML.load_file(Rails.root.join(".swagcov.yml"))
+      raise BadConfigurationError, "Invalid .swagcov.yml" unless valid?
```

Comment:
Love this! Possibly checking if `File.exist?(Rails.root.join(".swagcov.yml"))` before the `YAML.load` might be a good one as well. I don't insist on a refactoring PR.

Reply:
Since it's a `Pathname` — using [`#exist?`](https://ruby-doc.org/stdlib-2.6.3/libdoc/pathname/rdoc/Pathname.html#method-i-exist-3F)